src/main/java/rainier/org/csvcompare/CsvDiffApplication.java

```diff
@@ -0,0 +1,98 @@
+package rainier.org.csvcompare;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Calendar;
+import java.util.Map;
+import org.apache.http.client.HttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.log4j.Logger;
+import rainier.org.csvcompare.compare.CsvCompare;
+import rainier.org.csvcompare.compare.CsvCompareImp;
+import rainier.org.csvcompare.config.Config;
+import rainier.org.csvcompare.dto.FileToCompare;
+import rainier.org.csvcompare.loadFile.LoadFile;
+import rainier.org.csvcompare.loadFile.LoadFileHttp;
+import rainier.org.csvcompare.utils.CsvFileConverter;
+import rainier.org.csvcompare.utils.CsvFileConverterImp;
+
+public class CsvDiffApplication {
+
+  private static final Logger logger = Logger.getLogger(CsvDiffApplication.class);
+
+  public static void main(String[] args) {
+    CsvDiffApplication object = new CsvDiffApplication();
+    object.waitMethod();
+  }
+
+  private synchronized void waitMethod() {
+    while (true) {
+
+      try {
+        System.out.println(
+            "csv compare program is running ==> " + Calendar.getInstance().getTime());
+
+        CsvFileConverter<Map<String, String>> csvFileConverter = new CsvFileConverterImp();
+        HttpClient http = HttpClientBuilder.create().build();
+        LoadFile loadFile = new LoadFileHttp(http);
+        initCompare(CsvCompareImp.getInstance(), csvFileConverter, loadFile);
+        this.wait(Config.getTimeWindow());
+      } catch (InterruptedException ex) {
+        logger.error(ex.getMessage(), ex.getCause());
+        ex.printStackTrace();
+      } catch (IOException ex) {
+        int[] counter = new int[] {0};
```

Comment:
make this static class variable. you don't have to pass down to each function.

Reply:
sounds good, but this is just a one time function to implement the retry logic, if make a static class variable then that means I need to create a class for this?

Also since this is a counter, I need to reset it to zero when start a new retry here. The logic here is to retry maximum times which is 3 here to download the data. Each time when the retry function is called, I need to make sure the counter starts from 0. 

What do you think.